### PR TITLE
fix: only define Jenkins agent workdir as data volume (in the Jenkins home dir)

### PR DESCRIPTION
**Pull request by gounthar**
Context: this PR helps with #131.
It introduces the following changes:

- `/home/jenkins` is not a data volume anymore 
- `/home/jenkins/agent` is now a data volume
- `/home/jenkins/.jenkins` is now a data volume too
- README update
----

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- ~~Link to relevant pull requests, esp. upstream and downstream changes~~
- ~~Ensure you have provided tests - that demonstrates feature works or fixes the issue~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,24 +12,63 @@ See [Jenkins Distributed builds](https://wiki.jenkins-ci.org/display/JENKINS/Dis
...
+
+```bash
+docker run -v my-local-dir:path-on-the-jenkins-ssh-agent:rw jenkins/ssh-agent "<public key>"
+```
 
 ### How to use this image with Docker Plugin
 
-To use this image with [Docker Plugin](https://plugins.jenkins.io/docker-plugin), you need to
-pass the public SSH key using environment variable `JENKINS_AGENT_SSH_PUBKEY` and not as a startup argument.
+Même chose ici;, il faut décrire le remote file system root dans le docker template
```
Comment (dduportal) [inline README.md:41]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -12,24 +12,63 @@ See [Jenkins Distributed builds](https://wiki.jenkins-ci.org/display/JENKINS/Dis
...
+
+Should you need to extend the image, you could use something along those lines:
+
+```Dockerfile
+FROM jenkins/ssh-agent:bullseye-jdk17 as ssh-agent
+[...]
+RUN mkdir -p "${JENKINS_AGENT_HOME}"/.ssh
+COPY mykey "${JENKINS_AGENT_HOME}"/.ssh/mykey
+[...]
+```
```
Comment (dduportal) [inline README.md:71]:
````suggestion
```Dockerfile
FROM jenkins/ssh-agent:bullseye-jdk17 as ssh-agent
# [...]
COPY --chown=jenkins mykey "${JENKINS_AGENT_HOME}"/.ssh/mykey
# [...]
``` 
````

- Removing the `mkdir` RUN instruction: as per ````, `If <dest> doesn’t exist, it is created along with all missing directories in its path.`. 
- Added a chwon for the file copy to be sure it is owned by Jenkins user
- nitpick: Added a `#` to comment out the `[...]` line for better rendering with syntax color

File: README.md [hunk 3]
```diff
@@ -12,24 +12,67 @@ See [Jenkins Distributed builds](https://wiki.jenkins-ci.org/display/JENKINS/Dis
 
 ## Running
 
+### Running with the SSH Build Agents plugin
+
 To run a Docker container
 
 ```bash
-docker run jenkins/ssh-agent "<public key>"
+docker run -d --rm --name=agent -p 4444:22 -e "JENKINS_AGENT_SSH_PUBKEY=<public_key>" jenkins/ssh-agent
```
Comment (dduportal) [inline README.md:20]: nitpick: either use long or short flags for all

File: README.md [hunk 4]
```diff
@@ -12,24 +12,67 @@ See [Jenkins Distributed builds](https://wiki.jenkins-ci.org/display/JENKINS/Dis
...
+![Remote root directory with a Linux agent](docs/ssh-plugin-remote-root-directory-linux.png "Remote root directory with a Linux agent")
+
+When using the Windows image, you have to set the value of the `Remote root directory` to `C:/Users/jenkins/Work` in the agent configuration UI.
+
+![Remote root directory with a Windows agent](docs/ssh-plugin-remote-root-directory-windows.png "Remote root directory with a Windows agent")
+
+If you intend to use another directory than `/home/jenkins/agent` under Linux or `C:/Users/jenkins/Work` under Windows, don't forget to add it as a data volume.
+
+```bash
+docker run -v my-local-dir:path-on-the-jenkins-ssh-agent:rw jenkins/ssh-agent "<public key>"
```
Comment (dduportal) [inline README.md:43]:
```suggestion
docker run -v docker-volume-for-jenkins-ssh-agent:/home/jenkins/agent:rw jenkins/ssh-agent "<public key>"
```

- Prefer recommending a data volume by default 
- Is this snippet to show where to mount with the default mount path, or is it about specifying another one?

**Review by dduportal — APPROVED** [review]
Nice work! None of my last suggestion should block the PR: it is only question on documentation changes, which are off-topic.

Your work is clearly better than the current production!

File: README.md [hunk 5]
```diff
@@ -12,24 +12,67 @@ See [Jenkins Distributed builds](https://wiki.jenkins-ci.org/display/JENKINS/Dis
...
 ```bash
-docker run jenkins/ssh-agent "<public key>"
+docker run -d --rm --name=agent -p 4444:22 -e "JENKINS_AGENT_SSH_PUBKEY=<public_key>" jenkins/ssh-agent
 ```
 
-You'll then be able to connect this agent using the [SSH Build Agents plugin](https://plugins.jenkins.io/ssh-slaves) as "jenkins" with the matching private key.
+ - `-d`: To start a container in detached mode, use the `-d` option. Containers started in detached mode exit when the root process used to run the container exits, unless you also specify the --rm option.
+ - `--rm`: If you use -d with --rm, the container is removed when it exits or when the daemon exits, whichever happens first.
+ - `--name`: Assigns a name to the container
+ - `-p`: Publishes a container's port(s) to the host. Here, we publish the container SSH port 22 to the host port 4444.
```
[the remark this answers is not in the document]
Comment (gounthar) [reply README.md:26]: You're totally right about that one. 💯 
The thing is, the existing command in the documentation does not work (stupid copy/paste, you have to know a little bit of `docker` to add the right args to get it started).
I then started to add the ones I use, and so I added the `-p`. When doing it, I knew it looked weird, so I shouldn't have added it.

File: README.md (hunk 3 above)
Comment (gounthar) [reply README.md:20]: I'm sorry, but I don't know the long flag equivalent for [`-e`](https://docs.docker.com/engine/reference/run/#env-environment-variables), [`-d`](https://docs.docker.com/engine/reference/run/#detached--d) and [`-p`](https://docs.docker.com/engine/reference/run/#expose-incoming-ports).
I don't know either the short flag for [`--rm`](https://docs.docker.com/engine/reference/run/#clean-up---rm)  and [`--name`](https://docs.docker.com/engine/reference/run/#name---name). 😨

File: README.md (hunk 4 above)
Comment (gounthar) [reply README.md]: Would the [${AGENT_WORKDIR}](url) variable defined in the [`Dockerfile`](https://github.com/gounthar/docker-ssh-agent/blob/volume-one-liner/11/bullseye/Dockerfile#L62) still be visible?
I could totally be wrong though.... 🤔 
That would give something like:
`docker run -v docker-volume-for-jenkins-ssh-agent:${AGENT_WORKDIR}:rw jenkins/ssh-agent "<public key>"

Sorry, I'm muddying the water there...
My initial goal was to describe a specific new path, and not how to use the default one. 😊 
So please forget what I wrote a few lines above.
`

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (dduportal) [reply README.md]: I really think that your instinct is great: documenting the main use cases is good. 
It's only that adding the network use cases is off-topic for this PR, but it makes a lot of sense and value to do it!

File: README.md (hunk 4 above)
Comment (dduportal) [reply README.md]: > Would the [${AGENT_WORKDIR}](https://github.com/jenkinsci/docker-ssh-agent/pull/url) variable defined in the [Dockerfile](https://github.com/gounthar/docker-ssh-agent/blob/volume-one-liner/11/bullseye/Dockerfile#L62) still be visible?
> That would give something like:
> `docker run -v docker-volume-for-jenkins-ssh-agent:${AGENT_WORKDIR}:rw jenkins/ssh-agent ""

Only *inside* the container itself. In your example here, the variable would be resolved by the shell where the command is run.

While `docker run jenkins/ssh-agent env` would show you that `AGENT_WORKDIR` is in the container's environment.

>  My initial goal was to describe a specific new path, and not how to use the default one. 😊

Make sense. WDYT of something like that (adding the intent before the code snippet + a screenshot of the Jenkins UI with the correct value after the snippet):

````
For instance if you want to use `/workspace` as the agent working directory in the container mounted from a Docker volume named `docker-volume-for-jenkins-ssh-agent`

```bash
docker run --volume docker-volume-for-jenkins-ssh-agent:/workspace:rw jenkins/ssh-agent "<public key>"
```

and configure the "Remote Working Directory" to `/workspace`:

<screenshot here>
````

File: README.md (hunk 3 above)
Comment (gounthar) [reply README.md]: I'll do the modification, thanks for the tip. ❤️

**Pull request merged**